src/index.ts

```diff
@@ -43,6 +43,31 @@ type ExpandPropsOf<T extends Resource['Read'], U extends ODataOptions<T>> =
 			? StringKeyOf<U['$expand']>
 			: // If no $expand is provided, no properties are expanded
 				never;
+type ExpandToResponse<T extends Resource['Read'], U extends ODataOptions<T>> =
+	U['$expand'] extends ReadonlyArray<StringKeyOf<T>>
+		? {
+				[P in U['$expand'][number]]-?: Array<
+					PickDeferred<Expanded<T[P]>[number]>
+				>;
+			}
+		: U['$expand'] extends { [key in StringKeyOf<T>]?: any }
+			? {
+					[P in keyof U['$expand']]-?: OptionsToResponse<
+						Expanded<T[P & string]>[number],
+						NonNullable<U['$expand'][P]>
```

Comment:
@Page- I tested locally and I think this NonNullable can be gone (if I understand correctly, this might reduce the amount of needed `as const`)

Reply:
done